VMBackup/main/handle.py

```diff
@@ -172,31 +185,58 @@ def exit_if_same_taskId(taskId):
 def convert_time(utcTicks):
     return datetime.datetime(1, 1, 1) + datetime.timedelta(microseconds = utcTicks / 10)
 
-def set_do_seq_flag():
-    configfile='/etc/azure/vmbackup.conf'
-    try:
+def get_value_from_configfile(key):
+    global backup_logger
+    value = None
+    configfile = '/etc/azure/vmbackup.conf'
+    try :
+        if os.path.exists(configfile):
+            config = ConfigParser.ConfigParser()
+            config.read(configfile)
+            if config.has_option('SnapshotThread',key):
+                value = config.get('SnapshotThread',key)
+            else:
+                backup_logger.log("Config File doesn't have the key :" + key, True, 'Info')
+    except Exception as e:
+        errorMsg = " Unable to get config file.key is "+ key +"with error: %s, stack trace: %s" % (str(e), traceback.format_exc())
+        backup_logger.log(errorMsg, True, 'Warning')
+    return value
+
+def set_value_to_configfile(key, value):
+    configfile = '/etc/azure/vmbackup.conf'
+    try :
         backup_logger.log('setting doseq flag in config file', True, 'Info')
         if not os.path.exists(os.path.dirname(configfile)):
             os.makedirs(os.path.dirname(configfile))
-
+        config = ConfigParser.RawConfigParser()
         if os.path.exists(configfile):
-            config = ConfigParser.ConfigParser()
             config.read(configfile)
-            if not config.has_option('SnapshotThread','doseq'):
-                file_pointer = open(configfile, "a")
-                file_pointer.write("doseq: 1")
-                file_pointer.close()
-        else :
-            file_pointer = open(configfile, "w")
-            file_pointer.write("[SnapshotThread]\n")
-            file_pointer.write("doseq: 1")
-            file_pointer.close()
+            if config.has_section('SnapshotThread'):
+                if config.has_option('SnapshotThread', key):
+                    config.remove_option('SnapshotThread', key)
+                    config.set('SnapshotThread', key, value)
+                else:
+                    config.set('SnapshotThread', key, value)
+            else:
+                config.add_section('SnapshotThread')
+                config.set('SnapshotThread', key, value)
+        else:
+            config.add_section('SnapshotThread')
+            config.set('SnapshotThread', key, value)
```

Comment:
config.set('SnapshotThread', key, value)  is common for all if-elses, can be moved outside after if-elses

Reply:
done